File: csharp/bob/Bob.cs
public static class Bob
{
    public static string Response(string statement)
    {
        if (IsSilence(statement))
        {
            return "Fine. Be that way!";
        }

        if (IsAsking(statement) && IsShouting(statement))
        {
            return "Calm down, I know what I'm doing!";
        }

        if (IsAsking(statement))
        {
            return "Sure.";
        }

        if (IsShouting(statement))
        {
            return "Whoa, chill out!";
        }

        return "Whatever.";
    }

    private static bool IsShouting(string statement) => statement.Any(char.IsLetter) && !statement.Any(char.IsLower);
    private static bool IsAsking(string statement) => statement.Trim().EndsWith("?");
    private static bool IsSilence(string statement) => string.IsNullOrWhiteSpace(statement);
}

File: javascript/bob/bob.js
export const hey = (utterance) => {
  utterance = utterance.replace(/[\s]+$/g, '')

  if (isYelling(utterance) && isAsking(utterance)) { return 'Calm down, I know what I\'m doing!'}
  if (isYelling(utterance)) { return 'Whoa, chill out!' }
  if (isAsking(utterance)) { return 'Sure.'}
  if (isSilence(utterance)) { return 'Fine. Be that way!' }
  return 'Whatever.'
}

const isSilence = (statement) => statement.length == 0
const isYelling = (statement) => /[A-Z]/.test(statement) && statement.toUpperCase() == statement
const isAsking = (statement) => /\?$/.test(statement)